query-service-impl/src/test/java/org/hypertrace/core/query/service/AbstractServiceTest.java

```diff
@@ -0,0 +1,51 @@
+package org.hypertrace.core.query.service;
+
+import com.google.protobuf.GeneratedMessageV3;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.util.JsonFormat;
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.util.stream.Collectors;
+import org.hypertrace.core.grpcutils.context.RequestContext;
+
+public abstract class AbstractServiceTest<TQueryServiceRequestType extends GeneratedMessageV3> {
+
+  private static final String QUERY_SERVICE_TEST_REQUESTS_DIR = "test-requests";
+
+  protected TQueryServiceRequestType readQueryServiceRequest(String fileName) {
```

Comment:
What is `T`stands for? What about just having `QueryServiceRequestType` Is it how it was named in `proto`?

Reply:
I think it is to denote a generic class type. It is same as definer here by you - https://github.com/hypertrace/gateway-service/blob/1997536bcdddfc4d00aed3af7392fc1bb2ecdf91/gateway-service-impl/src/test/java/org/hypertrace/gateway/service/common/AbstractServiceTest.java#L45